### PR TITLE
build(deps) : standardise react and vite libs

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
| Question         | Answer
| ---------------- | ---
| Branch?          |`develop` 
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-14678
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Improve cds build react apps

### CODE REVIEW

File: packages/manager-components/package.json [hunk 1]
```diff
@@ -83,11 +82,11 @@
...
     "storybook-addon-react-router-v6": "^2.0.10",
     "ts-jest": "^29.1.1",
-    "typescript": "^4.3.2",
-    "vite": "4.4.9",
+    "vite": "^5.2.13",
     "vite-plugin-dts": "3.5.1",
     "vitest": "0.34.1",
-    "zustand": "^4.5.0"
+    "zustand": "^4.5.0",
+    "@ovh-ux/manager-vite-config": "^0.6.2"
```
Comment (anooparveti) [inline packages/manager-components/package.json:89]: **nitpick:** This should be in the correct sorting order. Strange that it was not auto-corrected by `sort-package-json`.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/pci-vouchers/package.json [hunk 2]
```diff
@@ -55,10 +51,18 @@
...
     "rollup": "^4.4.0",
-    "tailwindcss": "^3.4.0",
-    "typescript": "^5.3.3",
-    "vite": "^3.0.2",
-    "vitest": "^1.2.0"
+    "tailwindcss": "^3.4.0"
+  },
+  "peerDependencies": {
+    "date-fns": "workspace:*",
+    "i18next": "workspace:*",
```
Comment (frenautvh) [inline packages/manager/apps/pci-vouchers/package.json:58]: why put those dependencies in peerDeps ?

a micro app is providing it's own react version, what if we have 30 or more µapp ? we'll have to bump all the apps at the same ; same issue for other packages added in peer dep

File: packages/manager-components/package.json (hunk 1 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/package.json:89]: 👍

**Review by anooparveti — APPROVED** [review]

**Review by chipp972 — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

File: packages/manager-components/package.json [hunk 3]
```diff
@@ -8,16 +8,24 @@
...
+    "build": "echo build",
     "build:storybook": "storybook build",
+    "dev": "tsc",
     "prettier": "prettier --write \"src/**/*.{ts,tsx,js,mdx}\"",
+    "start": "storybook dev -p 6006",
     "test": "jest --coverage -u",
-    "test:watch": "jest --watch",
     "test:cov": "vitest run --coverage",
-    "test:e2e": "tsc && npx playwright test --ui --debug"
+    "test:e2e": "tsc && npx playwright test --ui --debug",
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/package.json:22]: yes we delete this in another PR 👍

File: package.json [hunk 4]
```diff
@@ -115,7 +116,7 @@
     "stylelint-prettier": "^1.1.2",
     "stylelint-scss": "^3.18.0",
     "turbo": "^1.4.3",
-    "vite": "^4.5.2",
+    "vite": "^5.2.13",
```
Comment (frenautvh) [inline package.json:119]: i don't think vite nor date-fns should be declared as dependencies at the root package.json of the monorepo

https://turbo.build/repo/docs/crafting-your-repository/managing-dependencies#install-dependencies-where-theyre-used

dependencies should be declared where they are used, the root package.json should ideally just contains the monorepo dependencies such as turbo / yarn / etc

**Review by anooparveti — APPROVED** [review]

**Pull request merged**